src/routes/items/[item_id]/+page.server.ts

```diff
@@ -27,17 +28,25 @@ export const actions = {
             var completeItem = {
                 ...{ "item_id": item.item_id }, ...inputItem
             };
+            // TODO: need to find a better way to do this
+            const imageChanged: boolean = !((data.get("image") as File).name === "undefined");
+            const itemChanged = hasItemBeenUpdated(completeItem, item);
 
-            if (isEqual(completeItem, item)) {
-                return fail(400, { "message": "No field selected to be updated -_^" })
+            if (!itemChanged && !imageChanged) {
+                return fail(400, { failure: true, errorMessage: "No field selected to be updated -_^" })
             }
 
             try {
-                await ItemsService.itemsItemIdPutItemsItemIdPut(item.item_id!, inputItem);
+                if (itemChanged) {
+                    await ItemsService.itemsItemIdPutItemsItemIdPut(item.item_id!, inputItem);
+                }
+                if (imageChanged) {
+                    await uploadImageForItem(item, data);
+                }
```

Comment:
why is this not part of the same UI?

Reply:
https://github.com/abhiandthetruth/wardrobe-api/wiki/Architecture-Decision-Records#separate-image-api-for-items